ddtrace/filters.py

```diff
@@ -1,9 +1,26 @@
+import abc
 import re
+from typing import List
+from typing import Optional
+
+from ddtrace import Span
+from ddtrace.vendor import six
 
 from .ext import http
 
 
-class FilterRequestsOnUrl(object):
+class TraceFilter(six.with_metaclass(abc.ABCMeta)):
+    @abc.abstractmethod
+    def process_trace(self, trace):
+        # type: (List[Span]) -> Optional[List[Span]]
+        """Processes a trace.
+
+        None can be returned to prevent the trace from being exported.
+        """
+        pass
```

Comment:
Would raising `NotImplementedError` help with not forgetting to implement the method in subclasses? It looks as if any subclass that is not implementing this method will effectively act like a discard-all filter.

Reply:
The ABC will handle this:

```python
>>> from ddtrace.filters import TraceFilter
>>> class Filter(TraceFilter):
...     pass
... 
>>> Filter()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: Can't instantiate abstract class Filter with abstract methods process_trace
```

although probably worth having a test case enforcing this behaviour so I'll add that!